src/tests/unit/doughs-dao.test.js

```diff
@@ -0,0 +1,85 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import sinon from 'sinon';
+
+import { getDoughs } from 'api/v1/db/oracledb/doughs-dao';
+import * as doughsSerializer from 'api/v1/serializers/doughs-serializer';
+import * as connectionApi from 'api/v1/db/oracledb/connection';
+// import { connection } from 'oracledb';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const baseGetDoughsReturn = {
+  data: [],
+};
+
+describe('test doughs dao', () => {
+  let serializeDoughsStub = null;
+  let getConnectionStub = null;
+  let connectionSpy = null;
+  const getDoughsConditions = 'ID AS "id", NAME AS "name", GRAMS_FLOUR AS "gramsFlour", GRAMS_WATER AS "gramsWater", FLOUR_TYPE AS "flourType", WATER_TEMP AS "waterTemp", GRAMS_YEAST AS "gramsYeast", GRAMS_SALT AS "gramsSalt", GRAMS_SUGAR AS "gramsSugar", GRAMS_OLIVE_OIL AS "gramsOliveOil", BULK_FERMENT_TIME AS "bulkFermentTime", PROOF_TIME AS "proofTime", SPECIAL_INSTRUCTIONS AS "specialInstructions"';
+  const getDoughsQuery = `SELECT ${getDoughsConditions} FROM DOUGHS `;
+  const getDoughsQueryWithWaterTemp = `SELECT ${getDoughsConditions} FROM DOUGHS WHERE WATER_TEMP = :waterTemp ;`;
+  const getDoughsQueryWithMultipleConditions = `SELECT ${getDoughsConditions} FROM DOUGHS WHERE WATER_TEMP = :waterTemp AND GRAMS_WATER = :gramsWater AND PROOF_TIME = :proofTime AND NAME = :name ;`;
+  const executeReturnRows = ['a', 'b', 'c'];
+  const executeReturn = { rows: executeReturnRows, metadata: 'ignore-this' };
+  beforeEach(() => {
+    connectionSpy = sinon.stub().returns(executeReturn);
+    serializeDoughsStub = sinon.stub(doughsSerializer, 'serializeDoughs');
+    getConnectionStub = sinon.stub(connectionApi, 'getConnection');
+    getConnectionStub.resolves({
+      execute: connectionSpy,
+      close: () => { },
+      commit: () => { },
+    });
+    serializeDoughsStub.returns(baseGetDoughsReturn);
+  });
```

Comment:
Clever, but you still need to connect to the real DB and stub it right? When running unit tests, we shouldn't touch the real DB instance in case we screw things up. In other words, the unit test shouldn't rely on the config file at all. The unit tests won't work if you run without a config file.

Reply:
The DB should be stubbed here:
```js
getConnectionStub = sinon.stub(connectionApi, 'getConnection');
    getConnectionStub.resolves({
      execute: connectionSpy,
      close: () => { },
      commit: () => { },
    });
```
which stops the `getConnection` method from returning a real database connection

Reply:
Yeah, I see what you mean. My main concern is you still imported the connection [at the very beginning here](https://github.com/osu-mist/pizza-api/blob/d29ca1a3f2c89af5a4f1f6d25a9c79cfb8e4b42e/src/tests/unit/doughs-dao.test.js#L7) with the real config data.

Reply:
The only side effects from the import are these lines though from the start of the file:
```js
const dbConfig = config.get('dataSources').oracledb;

process.on('SIGINT', () => process.exit());
oracledb.outFormat = oracledb.OBJECT;
oracledb.fetchAsString = [oracledb.DATE, oracledb.NUMBER];

/** Increase 1 extra thread for every 5 pools but no more than 128 */
const threadPoolSize = dbConfig.poolMax + (dbConfig.poolMax / 5);
process.env.UV_THREADPOOL_SIZE = threadPoolSize > 128 ? 128 : threadPoolSize;

/** Connection pool */
let pool;
```

And I don't see anything that would actually trigger it to connect to the database. I think you'd have to call the actual `getConnection` or `getPool` functions to do that. I'm also not totally sure how to mock `getConnection` without using this import statement. I've seen `proxyquire` used in other API code but since I'm using the es6 module import syntax I'm not sure if that will work here.

Reply:
Yes, using `proxyquire` should be the correct way. Here're some examples:

Terms API: https://github.com/osu-mist/terms-api/blob/b9acb21bb42fa5a483af15f514ea9017f4366c9f/src/tests/unit/terms-dao.test.js#L15-L35
OnBase API: https://github.com/osu-mist/onbase-api/blob/cd088d37d74986168a2a4e6cafc2db4f4ec58e19/src/tests/unit/onbase-dao.test.js#L26-L44
Students API: https://github.com/osu-mist/students-api/blob/2ad5a02bb99043bc7ae8ca2571fd836ea2a8a300/src/tests/unit/students-dao.test.js#L18-L32

Reply:
I added proxyquire in the latest pr 👍

Reply:
Ah yeah, that's from `doughs-dao` generating the base url internally instead of having the serializer generate the base url, which seems to be the more common practice. I'll fix that.